src/test/regress/sql/grant_on_schema_propagation.sql

```diff
@@ -0,0 +1,153 @@
+--
+-- GRANT_ON_SCHEMA_PROPAGATION
+--
+
+-- test grants are propagated when the schema is
+CREATE SCHEMA dist_schema;
+CREATE TABLE dist_schema.dist_table (id int);
+CREATE SCHEMA another_dist_schema;
+CREATE TABLE another_dist_schema.dist_table (id int);
+CREATE SCHEMA non_dist_schema;
+
+-- create roles on all nodes
+SELECT run_command_on_coordinator_and_workers('CREATE USER role_1');
+SELECT run_command_on_coordinator_and_workers('CREATE USER role_2');
+SELECT run_command_on_coordinator_and_workers('CREATE USER role_3');
+
+-- do some varying grants
+GRANT USAGE, CREATE ON SCHEMA dist_schema TO role_1 WITH GRANT OPTION;
+GRANT USAGE ON SCHEMA dist_schema TO role_2;
+SET ROLE role_1;
+GRANT USAGE ON SCHEMA dist_schema TO role_3 WITH GRANT OPTION;
+GRANT CREATE ON SCHEMA dist_schema TO role_3;
+GRANT CREATE, USAGE ON SCHEMA dist_schema TO PUBLIC;
+RESET ROLE;
+GRANT USAGE ON SCHEMA dist_schema TO PUBLIC;
+
+SELECT create_distributed_table('dist_schema.dist_table', 'id');
+SELECT create_distributed_table('another_dist_schema.dist_table', 'id');
+
+\c - - - :worker_1_port
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname = 'dist_schema';
```

Comment:
can we also execute this query on the coordinator so the reader of the output can verify the equality